static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,400 @@
+          if (_.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          tissue = n.nodes[0].value;
+        } else if (n.property === "proteinExpressionScore") {
+          score = n.nodes[0].value;
+        }
+      }
+      returnData.push({ name: tissue, value: score });
+    }
+    return returnData;
+  }
+  return [];
+}
+export function getProteinInteraction(
+  data: GraphNodes,
+  nodeName: string
+): InteractingProteinType[] {
+  // Protein Interaction to confidence score mapping.
+  if (!data) {
+    return [];
+  }
+  const returnData = [] as InteractingProteinType[];
+  const seen = new Set();
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "interactingProtein") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let protein_name = null;
+      let confidence_score = null;
+      let parent_protein = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "name") {
+          // check for null name values and move to next iteration if true
+          if (_.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          protein_name = n.nodes[0].value;
+          parent_protein = nodeName;
+        } else if (n.property === "confidenceScore") {
+          // not checking for empty values because if name exists, confidence score must exist
+          for (const n1 of n.nodes) {
+            for (const n2 of n1.neighbors) {
+              if (n2.property === "value") {
+                const num = Number(n2.nodes[0].value);
+                if (num <= 1) {
+                  confidence_score = num;
+                }
+              }
+            }
+          }
+        }
+      }
+      // checking for duplicates
+      if (!seen.has(protein_name)) {
+        returnData.push({
+          name: protein_name,
+          value: confidence_score,
+          parent: parent_protein,
+        });
+        seen.add(protein_name);
+      }
+    }
+    return returnData;
+  }
+  return [];
+}
+
+export function getDiseaseGeneAssoc(data: GraphNodes): ProteinNumData[] {
+  // Disease Gene Associations
+  if (!data) {
+    return [];
+  }
+  const returnData: ProteinNumData[] = [];
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "geneID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let score = null;
+      let disease = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        return [];
+      }
+      for (const n of node.neighbors) {
```

Comment:
nit: as mentioned in my last round of review, we don't want this much nesting

Reply:
Done!

Reply:
Are you referring to the "if" condition with "diseaseOntologyID" ?

Reply:
I mean, there's a lot of "if" statements here that you can clean up to make this whole piece of code less nested